### ISSUE TITLE
[Linux] Error message when Caps Lock key is pressed (debug mode)

### ISSUE THREAD

**Issue by HidenoriMatsubayashi** [flutter/flutter#63939]
I use the real keyboard. The Error message is showed when Caps lock key is pressed.
(No problem with macos.)

## Steps to Reproduce

1. $ git clone https://github.com/google/flutter-desktop-embedding.git
2. $ cd testbed
3. $ flutter build linux --debug
4. $ cd build/linux/debug/bundle
5. $ ./testbed
6. click "Test raw keyboard events"
7. press Caps lock key

![image](https://user-images.githubusercontent.com/62131389/88992430-5a8ea980-d31e-11ea-9e2a-12e98966852b.png)

```
flutter: ══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
flutter: The following assertion was thrown during a platform message callback:
flutter: Attempted to send a key down event when no keys are in keysPressed. This state can occur if the key
flutter: event being sent doesn't properly set its modifier flags. This was the event:
flutter: RawKeyDownEvent#67540(logicalKey: LogicalKeyboardKey#0ff30(keyId: "0x1060000ff30", keyLabel: null,
flutter: debugName: "Unknown key code 65328"), physicalKey: PhysicalKeyboardKey#70039(usbHidUsage:
flutter: "0x00070039", debugName: "Caps Lock")) and its data: RawKeyEventDataLinux(keyLabel: null, keyCode:
flutter: 65328, scanCode: 66, unicodeScalarValues: 0, modifiers: 8192, modifiers down: {})
flutter: 'package:flutter/src/services/raw_keyboard.dart':
flutter: Failed assertion: line 599 pos 12: '<optimized out>'
flutter: 
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new?template=BUG.md
flutter: 
flutter: When the exception was thrown, this was the stack:
flutter: #2      RawKeyboard._handleKeyEvent (package:flutter/src/services/raw_keyboard.dart:599:12)
flutter: #3      BasicMessageChannel.setMessageHandler.<anonymous closure> (package:flutter/src/services/platform_channel.dart:74:49)
flutter: #4      _DefaultBinaryMessenger.handlePlatformMessage (package:flutter/src/services/binding.dart:283:33)
flutter: #5      _invoke3.<anonymous closure> (dart:ui/hooks.dart:280:15)
flutter: #9      _invoke3 (dart:ui/hooks.dart:279:10)
flutter: #10     _dispatchPlatformMessage (dart:ui/hooks.dart:154:5)
flutter: (elided 5 frames from class _AssertionError and dart:async)
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
```

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.21.0-10.0.pre.114, on Linux, locale ja_JP.UTF-8)
    • Flutter version 1.21.0-10.0.pre.114 at /opt/flutter
    • Framework revision b2c0970aa7 (23 hours ago), 2020-08-16 06:11:02 -0400
    • Engine revision b300be3df3
    • Dart version 2.10.0 (build 2.10.0-34.0.dev)

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1
```

</details>

**Comment by TahaTesser** [comment]
Hi @HidenoriMatsubayashi 
Not able to reproduce this issue on linux on latest `master` using https://github.com/google/flutter-desktop-embedding.git

![image](https://user-images.githubusercontent.com/48603081/90488273-4e497f80-e159-11ea-8b6d-7163dcd2abe2.png)

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.21.0-10.0.pre.139, on Linux, locale en_US.UTF-8)
    • Flutter version 1.21.0-10.0.pre.139 at /home/taha/Code/flutter
    • Framework revision a51dc3d913 (85 minutes ago), 2020-08-18 02:51:02 -0400
    • Engine revision 0b330467da
    • Dart version 2.10.0 (build 2.10.0-38.0.dev)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/taha/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /home/taha/Code/sdk
    • Java binary at: /home/taha/Code/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.
[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1
[✓] Android Studio (version 4.0)
    • Android Studio at /home/taha/Code/android-studio
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
[✓] VS Code (version 1.48.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.13.2
[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Linux
• No issues found!
```

</details> 

I cannot reproduce this issue on latest `master` channel.
Can you upgrade to `master` and try again
`flutter channel master`
`flutter upgrade --force`
`flutter doctor -v`

**Comment by HidenoriMatsubayashi** [comment]
@TahaTesser 

I tried again using latest master channel and could reproduce the issue. However, after trying it a few times, I was unable to reproduce it. I'll investigate the details next time I reproduce.

```
$ flutter doctor -v
[✓] Flutter (Channel master, 1.21.0-10.0.pre.139, on Linux, locale ja_JP.UTF-8)
    • Flutter version 1.21.0-10.0.pre.139 at /opt/flutter
    • Framework revision a51dc3d913 (2 hours ago), 2020-08-18 02:51:02 -0400
    • Engine revision 0b330467da
    • Dart version 2.10.0 (build 2.10.0-38.0.dev)

(snip) 

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

(snip) 

! Doctor found issues in 2 categories.
```

**Comment by officer47p** [comment]
The same thing happens on flutter web channel beta 1.20.2.
Pressing Caps lock produces an error.

**Javascript error(development tools, console) :**
`5[Intervention] Slow network is detected. See <URL> for more details. Fallback font will be used while loading: <URL>
js_primitives.dart:30 ══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
js_primitives.dart:30 The following assertion was thrown during a platform message callback:
js_primitives.dart:30 Assertion failed: file:///D:/flutter/packages/flutter/lib/src/widgets/shortcuts.dart:77:16
js_primitives.dart:30 keys.isNotEmpty
js_primitives.dart:30 is not true
js_primitives.dart:30 
js_primitives.dart:30 When the exception was thrown, this was the stack:
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 214:49  throw_
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 24:3    assertFailed
js_primitives.dart:30 packages/flutter/src/widgets/shortcuts.dart 77:20                                                                          fromSet
js_primitives.dart:30 packages/flutter/src/widgets/shortcuts.dart 181:63                                                                         fromSet
js_primitives.dart:30 packages/flutter/src/widgets/shortcuts.dart 309:63                                                                         handleKeypress
js_primitives.dart:30 packages/flutter/src/widgets/shortcuts.dart 472:20                                                                         [_handleOnKey]
js_primitives.dart:30 packages/flutter/src/widgets/focus_manager.dart 1623:43                                                                    [_handleRawKeyEvent]
js_primitives.dart:30 packages/flutter/src/services/raw_keyboard.dart 598:68                                                                     _handleKeyEvent
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54            runBody
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:12           _async
js_primitives.dart:30 packages/flutter/src/services/raw_keyboard.dart 562:34                                                                     [_handleKeyEvent]
js_primitives.dart:30 packages/flutter/src/services/platform_channel.dart 76:49                                                                  <fn>
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54            runBody
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:12           _async
js_primitives.dart:30 packages/flutter/src/services/platform_channel.dart 75:57                                                                  <fn>
js_primitives.dart:30 packages/flutter/src/services/binding.dart 267:33                                                                          handlePlatformMessage
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54            runBody
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:12           _async
js_primitives.dart:30 packages/flutter/src/services/binding.dart 258:37                                                                          handlePlatformMessage
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/window.dart 774:13                         _invoke3
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/window.dart 442:5                          invokeOnPlatformMessage
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/keyboard.dart 125:12                       [_handleHtmlEvent]
js_primitives.dart:30 C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/keyboard.dart 39:7                         <fn>
js_primitives.dart:30 ════════════════════════════════════════════════════════════════════════════════════════════════════`

**Flutter console itself:**
`══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
The following assertion was thrown during a platform message callback:
Assertion failed: file:///D:/flutter/packages/flutter/lib/src/widgets/shortcuts.dart:77:16
keys.isNotEmpty
is not true

When the exception was thrown, this was the stack:
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart
214:49  throw_
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 24:3
assertFailed
packages/flutter/src/widgets/shortcuts.dart 77:20
fromSet
packages/flutter/src/widgets/shortcuts.dart 181:63
fromSet
packages/flutter/src/widgets/shortcuts.dart 309:63
handleKeypress
packages/flutter/src/widgets/shortcuts.dart 472:20
[_handleOnKey]
packages/flutter/src/widgets/focus_manager.dart 1623:43
[_handleRawKeyEvent]
packages/flutter/src/services/raw_keyboard.dart 598:68
_handleKeyEvent
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54
runBody
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:12
_async
packages/flutter/src/services/raw_keyboard.dart 562:34
[_handleKeyEvent]
packages/flutter/src/services/platform_channel.dart 76:49
<fn>
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54
runBody
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:12
_async
packages/flutter/src/services/platform_channel.dart 75:57
<fn>
packages/flutter/src/services/binding.dart 267:33
handlePlatformMessage
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54
runBody
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:12
_async
packages/flutter/src/services/binding.dart 258:37
handlePlatformMessage
C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/window.dart 774:13
_invoke3
C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/window.dart 442:5
invokeOnPlatformMessage
C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/keyboard.dart 125:12
[_handleHtmlEvent]
C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/keyboard.dart 39:7
<fn>
════════════════════════════════════════════════════════════════════════════════════════════════════
`

**Comment by TahaTesser** [comment]
I just tried again, couldn't reproduce any error message 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-10.0.pre.87, on Linux, locale en_US.UTF-8)
    • Flutter version 1.22.0-10.0.pre.87 at /home/taha/Code/flutter_master
    • Framework revision 4732a214a7 (2 days ago), 2020-09-04 21:05:02 -0400
    • Engine revision ac8b9c4c52
    • Dart version 2.10.0 (build 2.10.0-86.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/taha/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /home/taha/Code/sdk
    • Java binary at: /home/taha/Code/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 4.0)
    • Android Studio at /home/taha/Code/android-studio
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.14.1

[✓] Connected device (3 available)
    • Linux (desktop)  • linux      • linux-x64      • Linux
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome
      85.0.4183.83

• No issues found!

```

</details> 

given your last message I feel safe to close this issue, if you disagree please write in the comments and I will reopen it. 
Thank you

**Comment by Hixie** [comment]
cc @gspencergoog

**Comment by HidenoriMatsubayashi** [comment]
I can reproduce the issue by using Flutter SDK version 1.22.-0-10.0.pre.309. Therefore, I'll check details later.

**Comment by gspencergoog** [comment]
I think this was fixed with #69576
